src/v2/Components/ArtworkFilter/ArtworkFilters/ArtistsFilter.tsx

```diff
@@ -21,11 +21,13 @@ const ArtistItem: React.FC<{
   name: string
   followedArtistSlugs: string[]
   isFollowedArtistCheckboxSelected: boolean
+  CheckboxProps: CheckboxProps
```

Comment:
You'd want to extend or intersect with `CheckboxProps` and then just spread `...rest` — Rather than trying to encapsulate the props in another prop.

```tsx
{  slug: string  name: string  followedArtistSlugs: string[]  isFollowedArtistCheckboxSelected: boolean } & CheckboxProps
```

```tsx
({  slug,  name,  followedArtistSlugs,  isFollowedArtistCheckboxSelected,  ...rest }) => {
```

```tsx
    <Checkbox
      selected={
        // @ts-expect-error STRICT_NULL_CHECK
        currentlySelectedFilters().artistIDs.includes(slug) ||
        (isFollowedArtistCheckboxSelected && isFollowedArtist)
      }
      onSelect={selected => {
        return toggleArtistSelection(selected, slug)
      }}
      {...rest}
    >
```

-----

Alternatively — it'd be also fine to just recall the theme config within this component itself.

Reply:
I like the idea of incapsulating props, thank you